test/integration/enumerate_databases/enumerate_databases.prose.test.ts

```diff
@@ -0,0 +1,71 @@
+import { expect } from 'chai';
+
+import type { MongoClient } from '../../../src';
+
+const REQUIRED_DBS = ['admin', 'local', 'config'];
+const DB_NAME = 'listDatabasesTest';
+
+describe('listDatabases() spec prose', function () {
+  /**
+   * Execute the method to enumerate full database information (e.g. listDatabases())
+   * - Verify that the method returns an Iterable of Document types
+   * - Verify that all databases on the server are present in the result set
+   * - Verify that the result set does not contain duplicates
+   */
+  let client: MongoClient;
+  let ENTIRE_DB_LIST: string[];
+
+  beforeEach(async function () {
+    client = this.configuration.newClient();
+    await client.connect();
+
+    const dbInfoFromCommand = await client.db('admin').command({ listDatabases: 1 });
+    const databasesToDrop = dbInfoFromCommand.databases
+      .map(({ name }) => name)
+      .filter(name => !REQUIRED_DBS.includes(name));
+
+    for (const dbToDrop of databasesToDrop) {
+      await client.db(dbToDrop).dropDatabase();
+    }
+
+    await client.db(DB_NAME).createCollection(DB_NAME);
+
+    ENTIRE_DB_LIST = (await client.db('admin').command({ listDatabases: 1 })).databases.map(
+      ({ name }) => name
+    );
+    ENTIRE_DB_LIST.sort();
```

Comment:
Should we add an expectations here around the ENTIRE_DB_LIST not being empty?

Reply:
sure!